### PR TITLE
Added new Share Box

### PR DESCRIPTION
**Pull request by markbao**
Added a shiny new Share Box.

- Synced up with the share photos on the TWS Facebook page.
- Snazzy animation.
- Supports mobile.

@ankitshah811: Customization instructions in inline commit comments.

screen shot 2015-05-14 at 6 10 54 am

### CODE REVIEW

File: app/views/cities/forbes_show.html.erb [hunk 1]
```diff
@@ -30,11 +30,9 @@
   	<% end %>
     <% if @city.unapproved? %>
     <% else %>
-      <div class='sharing-info'>
-        <%= render partial: 'shared/sharing_widget', locals: {city: @city} %>
-      </div>
     <% end %>
     -->
+    <%= render partial: 'shared/sharing_widget', locals: {city: @city} %>
```
Comment (markbao) [inline app/views/cities/forbes_show.html.erb]: Not using `city` local (yet).

Can add a custom fallback text here (e.g. for Twitter).

File: app/views/shared/_sharing_widget.html.erb [hunk 2]
```diff
@@ -1,2 +1,51 @@
...
+  <h2>Spread the word</h2>
+  <p>You give strangers a good rep. You’re the type that asks “How are you?” and means it. You opt for high fives over handshakes. And the friends you attract are similarly awesome. They’ll thank you when you tell them about TWS.</p>
+</div>
+
+<div class="sharing-box">
+
+  <div class="sharing-preview sharing-preview-bounce" id="sharing-preview">
+    <%= image_tag 'landing/landing-graphic-redesign.jpg', id: 'sharing-preview-image' %>
+    <div class="sharing-preview-text"><span id="sharing-preview-text-content">Tea With Strangers is all about making meaningful interactions more fun.</span> // <a href="http://teawithstrangers.com/">teawithstrangers.com</a></div>
+  </div>
```
Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: Although this is included in SHARE_COLLECTION, this is duplicated here for performance reasons.

File: app/views/shared/_sharing_widget.html.erb [hunk 3]
```diff
@@ -1,2 +1,51 @@
...
+
+  <div class="sharing-try-blurb">This is easy to post, but if you want to get more creative, we have some other ideas.</div>
+
+  <a href="#" class="sharing-try-button"><button class="button"><%= image_tag 'shuffle.png' %> <span class="sharing-try-button-text">Try this on for size</span></button></a>
+
+  <div class="sharing-elsewhere">
+    <div class="sharing-elsewhere-text">Wanna share somewhere else?</div>
+    <div class="sharing-elsewhere-buttons">
+    
+      <a href="mailto:?subject=Tea%20With%20Strangers&body=Tea%20With%20Strangers%20is%20all%20about%20making%20our%20cities%20feel%20more%20like%20neighborhoods.%20We%27re%20more%20%22connected%22%20than%20ever%20before%2C%20but%20we%27re%20also%20more%20alone.%20And%20all%20we%20want%20to%20do%20is%20bring%20people%20together%20because%2C%20well%2C%20the%20world%20is%20better%20that%20way.%0A%0ACheck%20us%20out%20at%20http%3A//teawithstrangers.com/" class="sharing-email"><%= image_tag 'share/email.png' %></a>
```
Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: Customize: Email: The email mailto should be edited with your copy. Use this builder: http://mikemustaine.com/mmtools/files/mailTo.html

File: app/views/shared/_sharing_widget.html.erb [hunk 4]
```diff
@@ -1,2 +1,51 @@
...
+
+      <a href="https://twitter.com/intent/tweet?text=Tea+With+Strangers+http%3A%2F%2Fteawithstrangers.com%2F" class="sharing-twitter sharing-button twitter-share-button"><%= image_tag 'share/twitter.png' %></a>
```
Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: Customize: Twitter: The Twitter text should be customized (presently "Tea With Strangers http://teawithstrangers.com").

Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: TODO: Consider using a fallback for this (like "I've signed up to be notified when Tea With Strangers reaches Portland!")

Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: TODO: Alternatively consider using the blurbs from the images here.

File: app/views/shared/_sharing_widget.html.erb [hunk 5]
```diff
@@ -1,2 +1,51 @@
...
+
+      <a href="http://tumblr.com/widgets/share/tool?canonicalUrl=http://teawithstrangers.com/&title=Tea%20With%20Strangers" class="sharing-tumblr sharing-button"><%= image_tag 'share/tumblr.png' %></a>
```
Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: Customize: Tumblr: The Tumblr share is highly customizable. We can even put the same images for the Facebook as photo post types here. Might want to consider that. For now, though, customize the `&title` and `&content` and any other details as listed in the Tumblr Share documentation: https://www.tumblr.com/docs/en/share_button

File: app/views/shared/_sharing_widget.html.erb [hunk 6]
```diff
@@ -1,2 +1,51 @@
...
+SHARE_COLLECTION[6] = {img: '<%= image_url 'share/share-6.png' %>', text: 'Real people have #nofilter', url: 'https://www.facebook.com/teawithstrangers/photos/a.818701241549790.1073741844.542640869155830/818701584883089/?type=3' }
+SHARE_COLLECTION[7] = {img: '<%= image_url 'share/share-7.png' %>', text: 'Good vibes only', url: 'https://www.facebook.com/teawithstrangers/photos/a.818701241549790.1073741844.542640869155830/818701578216423/?type=3' }
+SHARE_COLLECTION[8] = {img: '<%= image_url 'share/share-8.png' %>', text: 'Look around', url: 'https://www.facebook.com/teawithstrangers/photos/a.818701241549790.1073741844.542640869155830/818701561549758/?type=3' }
+SHARE_COLLECTION[9] = {img: '<%= image_url 'share/share-9.png' %>', text: 'What\'s your story?', url: 'https://www.facebook.com/teawithstrangers/photos/a.818701241549790.1073741844.542640869155830/818701548216426/?type=3' }
+SHARE_COLLECTION[10] = {img: '<%= image_url 'share/share-10.png' %>', text: 'Hands were made for high-fiving', url: 'https://www.facebook.com/teawithstrangers/photos/a.818701241549790.1073741844.542640869155830/818701534883094/?type=3' }
```
Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: Customize: The formatting of this is really straightforward.

````javascript
{img: 'image.url', text: 'blurb text', url: 'Facebook photo URL'}
````

Don't screw it up, Ankit.

Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: TODO: Consider using Array.push to clean this up a bit, if we don't really need to keep things in the order specified.

File: app/views/shared/_sharing_widget.html.erb (hunk 2 above)
Comment (markbao) [inline app/views/shared/_sharing_widget.html.erb]: TODO: @ankitshah811 Decide on new image and set that as the `og:image` in the header as well as the default image  here.

File: app/views/cities/forbes_show.html.erb (hunk 1 above)
Comment (mehulkar) [inline app/views/cities/forbes_show.html.erb]: let's get rid of this right now. we want your beautiful code but don't want to use it yet

File: app/assets/javascripts/share.js [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    e.preventDefault();
+  })
+}
+
+// Popup open function
+var sharePopup = function(url) {
+  // Calculate screen dimensions to put window at center
+  var left = (screen.width/2)-(600/2);
+  var top = (screen.height/2)-(400/2);
+
```
Comment (mehulkar) [inline app/assets/javascripts/share.js]: lol. did you give up on css here?

Comment (markbao) [inline app/assets/javascripts/share.js]: Hah. This is actually for window.open, so it's not a dialog we can style with CSS (to the best of my knowledge). But still, I would've done this if I had to center with CSS.

**Pull request merged**